app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -677,10 +677,12 @@ public function getPublishableKeyForRoute()
         $checkoutType = static::CHECKOUT_TYPE_MULTI_PAGE;
         if ($routeName === 'adminhtml') {
             $checkoutType = static::CHECKOUT_TYPE_ADMIN;
-        } else if ( ($routeName === 'firecheckout') || ($routeName === 'checkout' && $controllerName === 'onepage') ) {
+        } else if ($routeName === 'checkout' && $controllerName === 'onepage') {
             $checkoutType = static::CHECKOUT_TYPE_ONE_PAGE;
         }
 
+        Mage::dispatchEvent('bolt_checkout_type_for_route', array('checkout_type' => $checkoutType));
```

Comment:
I like the idea of an event here.  Two things:

1.) We have a pre-built construct here that would be more appropriate in this circumstance that will allow us to work with checkouts that we will not be official supporting.  https://github.com/BoltApp/bolt-magento1/blob/e03a9507faaca49109b15c4862867ba95bda8f93/app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php#L162-L171

You would then use this in place of `Mage::dispactchEvent` and your return as follows:

```
return $this->getPublishableKey(
     $this->boltHelper()->doFilterEvent(
          'bolt_boltpay_filter_checkout_type',
          $checkoutType, 
          ['requested_path' => $requestedPath] # optionally, you could just pass $requestedPath w/o the array
     )
);
```

2.) We now have a very strict process and standard for the addition of any newly introduced events. 
https://www.notion.so/boltteam/Plugin-customization-guide-46d894ba914046da8cd0c533f4644385